### PR TITLE
Rewrite Installing CKAN with Docker Compose docs

### PR DESCRIPTION
**Pull request by kowh-ai**
Fixes https://github.com/ckan/ckan/issues/7262

### Proposed fixes:
With the new CKAN Docker repo (https://github.com/ckan/ckan-docker) going live we need to update the CKAN Install using Docker Compose docs to reflect these changes

### Features:

- [ ] includes tests covering changes
- [x ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by EricSoroos** [comment]
#6861 is an initial rewrite of `docker-compose` into `docker compose`. 

I'd recommend a (small) mention of `docker-compose` as v1, `docker compose` as v2, and prefer v2 in the docs but note that they're relatively interchangeable at this point.

File: doc/maintaining/installing/install-from-docker-compose.rst [hunk 1]
```diff
@@ -3,28 +3,22 @@
...
 
 --------------
 1. Environment
 --------------
-This tutorial was tested on Ubuntu 20.04 LTS.
-The hosts can be local environments or cloud VMs. It is assumed that the user has direct access
-(via terminal / ssh) to the systems and root permissions.
+The hosts can be local environments or cloud VMs. This tutorial works perfectly on Ubuntu 20.04 LTS (Focal) 
+and MacOS 11.5 (Big Sur). Other hosts should be fine also. It is assumed that the user has direct access 
+(via terminal / ssh) to the systems and has root access.
```
Comment (amercader) [inline doc/maintaining/installing/install-from-docker-compose.rst:21]: Let's mention the `docker-compose` vs `docker compose` difference in the Docker section below like @EricSoroos suggested, maybe with a `..note::` tag. We can link to [this page](https://docs.docker.com/compose/compose-v2/) and stress that it should be a drop-in replacement

Seeing how `docker-compose` has been deprecated for a while and be removed in June 2023 I think we should future proof the docs and use `docker compose` in the instructions (but keeping the note for users still using the old version)

Comment (amercader) [inline doc/maintaining/installing/install-from-docker-compose.rst:21]: I personally think that the "a. Storage" section below is out of scope for this repo and I'd lean towards deleting it. If you think that's useful we can keep it in another location

**Review by amercader — COMMENTED** [review]
Great work @kowh-ai , I added some comments after reviewing.

I'm still not sure where to keep this documentation, in docs.ckan.org or in the ckan-docker repo (linked from docs.ckan.org of course). What do you think?

Any thoughts @ckan/core ?

**Comment by pdelboca** [comment]
@amercader I would keep documentation in the correspondent repository (and linked in our docs.)

File: doc/maintaining/installing/install-from-docker-compose.rst [hunk 2]
```diff
@@ -311,25 +233,21 @@ For maintainers, the process is in summary:
   Alternatively, add a ``pip install`` step for the extension into a custom CKAN Dockerfile.
 * Restart ``ckan`` service, read logs.
 
-a. Download and install extension from inside ``ckan`` container into ``docker_ckan_home`` volume
+a. Download and install extension from inside ``ckan`` container into ``ckan-docker_ckan_home`` volume
```
[the remark this answers is not in the document]
Comment (kowh-ai) [reply doc/maintaining/installing/install-from-docker-compose.rst:236]: Yes it's handy to faff about in (running) containers to see immediate results (after a container restart) but maintainers may be happier working on a [Dockerfile](https://github.com/ckan/ckan-docker/blob/master/ckan/Dockerfile.dev)

[the remark this answers is not in the document]
Comment (kowh-ai) [reply doc/maintaining/installing/install-from-docker-compose.rst:236]: I'll revisit this section

**Comment by kowh-ai** [comment]
@amercader - OK after the meeting today I'll update this doc to be just a high level set of instructions with a link through to the  [ckan-docker](https://github.com/ckan/ckan-docker) repo which I'll add relevant parts of the old "Installing CKAN with Docker Compose" docs to...

**Comment by amercader** [comment]
Can you fix the conflicts @kowh-ai

**Comment by kowh-ai** [comment]
@amercader - updated `install-from-docker-compose.rst` so now all checks have passed. Trying to work out how to "Resolve conflicts" now...

**Comment by amercader** [comment]
Don't worry @kowh-ai, I'll take care of the conflicts, thanks!

**Pull request merged**